samples: sensor: accel_polling: Add test configuration for for nrf54h20 and nrf54l15

nordic-segl opened the pull request:
Add test configurations that allow to run:
- samples/sensor/accel_polling on nrf54h20 with PCA63566 SHIELD,
- samples/sensor/accel_polling on nrf54l15 with PCA63565 SHIELD,
- samples/sensor/bme680 on nrf54l15 with PCA63565 SHIELD.

nordic-segl committed:
samples: sensor: accel_polling: Add test configuration for nrf54h20

Add configuration that allows to run this sample on
the nrf54h20 target.
This requires SHIELD PCA63566 to be mounted on top of
the nrf54h20 development board PCA10175.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl commented on samples/sensor/accel_polling/boards/nrf54h20dk_nrf54h20_cpuapp.overlay:
```diff
@@ -0,0 +1,71 @@
+/ {
+	aliases {
+		accel0 = &adxl362;
+//		accel0 = &bmi270;
```
I was unable to get bmi270 working. Nor on nrf54l, nor nrf54h.
Maybe Kconfig is missing or driver requires sensor interrupt to be connected to a GPIO...

nordic-segl commented on samples/sensor/accel_polling/sample.yaml:
```diff
@@ -25,3 +25,18 @@ tests:
+    filter: dt_alias_exists("accel0")
+    harness: console
+    harness_config:
+      fixture: PCA63566
```
Not sure if sensor board marking (PCA63565 / PCA63566) shall be used or just 'sensor_adxl362'.

nordic-segl replied:
I've decided to switch fixture name to sensor_adxl362. In that case same test can be used for nrf54h and nrf54l targets while different PCA6356x have to be fitted on top of Development Kits.

nordic-segl replied:
obsolete

nordic-segl committed:
samples: sensor: accel_polling: Add test configuration for nrf54l15

Add configuration that allows to run this sample on
the nrf54l15 target.
This requires PCA63565 to be mounted on top of
the nrf54l15 preview development board PCA10156.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

kartben reviewed (changes requested):
similar to https://github.com/zephyrproject-rtos/zephyr/pull/71812, does this really need to be an overlay? Note how this sample has no overlay and how e.g Thingy52 and Thingy53 define their sensors right in their main DTS

nordic-segl commented:
> similar to #71812, does this really need to be an overlay? Note how this sample has no overlay and how e.g Thingy52 and Thingy53 define their sensors right in their main DTS

Both nrf54l and nrf54h need external add-on board with ADXL362 sensor. Thingy52/53 have that sensor mounted "inside".
Thus, I prefer NOT to add that node to the target DTS.

Another option is to define two SHIELDs (PCA63565 and PCA63566) like here:
https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/shields
Unfortunately, I don't have knowledge on how to combine Twister with SHIELD, thus use of overlay is simpler for me.
I don't know if it is possible to indicate SHIELD in the hardware_map.yaml in a way that Twister will run kernel tests on any nrf54l15 and accel_polling sample on nrf54l15+PCA63565.
I need to consult this...

nordic-segl committed:
samples: sensor: bme680: Add test configuration for nrf54l15

Add configuration thet allows to run this sample on
the nrf54l15 target.
This requires PCA63565 to be mounted on top of
the nrf54l15 preview development board PCA10156.

Signed-off-by: Sebastian Głąb <sebastian.glab@nordicsemi.no>

nordic-segl commented:
> > Both nrf54l and nrf54h need external add-on board with ADXL362 sensor. Thingy52/53 have that sensor mounted "inside". Thus, I prefer NOT to add that node to the target DTS.
> > Another option is to define two SHIELDs (PCA63565 and PCA63566) like here: https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/shields
> 
> Ya i missed the "This requires PCAxxxxx to be mounted on top of it." part in your initial comment. In this case yes, it should definitely be shields, as this will also make them much easier for folks to actually re-use vs. having to know that they should look for an overlay buried in some random code sample. I am not sure either about "testing" shields using Twister but maybe @erwango can comment on that

I just confirmed that PCA63565 and PCA63566 are for internal use only.
Because of this, I prefer NOT to define a new SHIELD. It can't be re-used by others.

Looking at the Twister documentation I see
https://docs.zephyrproject.org/latest/develop/test/twister.html

**fixture: <expression>**

    Specify a test case dependency on an external device(e.g., sensor), and identify setups that fulfill this dependency. It depends on specific test setup and board selection logic to pick the particular board(s) out of multiple boards that fulfill the dependency in an automation setup based on fixture keyword. Some sample fixture names are i2c_hts221, i2c_bme280, i2c_FRAM, ble_fw and gpio_loop.

Which seems to fit my use case.

nordic-segl commented on samples/sensor/accel_polling/boards/nrf54l15pdk_nrf54l15_cpuapp.overlay:
```diff
@@ -0,0 +1,57 @@
+/ {
+	aliases {
+		accel0 = &adxl362;
+//		accel0 = &bmi270;
```
Get it working or remove commented out nodes.

kartben commented:
> I just confirmed that PCA63565 and PCA63566 are for internal use only. Because of this, I prefer NOT to define a new SHIELD. It can't be re-used by others.

Then please do address this downstream. There is literally no other board adding such overlay in this sample so it's really hard to justify why we should start adding some, let alone ones that are vendor-specific. I think you would agree that it probably clutter the tree quite a bit if every vendor/user would start committing their custom overlays upstream :)

nordic-segl commented:
Main goal of this PR is to test SPI/I2C driver with a physical sensor at Zephyr upstream codebase. Driver tests based on GPIO loopbacks were not sufficient to detect errors.

SHIELD pca63565 is added in this PR: https://github.com/zephyrproject-rtos/zephyr/pull/71997
pca63566 will be added soon.

This solution was confirmed to work locally.
- fixture: pca63565 informs Twister that this test configuration can be run only on a Development Kit that has that fixture listed in the hardware_map.yaml
- extra_args: SHIELD=pca63565 adds sensor nodes to the DTS.

nordic-segl commented:
Moved to downstream.
